### PR TITLE
fix: Fixed preprocessor and added new DBNet pretrained params

### LINKED ISSUE #256: [transforms] Failing unittests for transformation
The following CI job is failing: https://github.com/mindee/doctr/pull/254/checks?check_run_id=2565047900
The reason behind that might either be an edge case or because of the modifications from #249

### PR DESCRIPTION
**Pull request by fg-mindee**
This PR introduces the following modifications:
- fixed a wrong normalization in the preprocessor introduced in #25
- refactored preprocessors with `doctr.transforms` for a cleaner interface
- added symmetric pad option in `transforms.Resize` and updated the repr
- updated predictors' repr
- added an image existence check in the constructor of datasets
- added a box flatness check in OCRDataset constructor
- added a new pretrained DBNet checkpoint
- updates the benchmark in the documentation
- updated unittests accordingly
- added recognition batch size as script argument for evaluation script
- fixed `transforms.RandomSaturation`

Briefly, with the fix + refactor of the preprocessor, the repr of the predictor is now much more comprehensive:
```python
from doctr.models import ocr_predictor
print(ocr_predictor(pretrained=True))
```
yields
```
OCRPredictor(
  (det_predictor): DetectionPredictor(
    (pre_processor): PreProcessor(
      (resize): Resize(output_size=(1024, 1024), method='bilinear')
      (normalize): Compose(
        (transforms): [
          LambdaTransformation(),
          Normalize(mean=[0.7979999780654907, 0.7850000262260437, 0.7720000147819519], std=[0.2639999985694885, 0.27489998936653137, 0.28700000047683716]),
        ]
      )
    )
    (model): DBNet(
      (feat_extractor): IntermediateLayerGetter()
      (fpn): FeaturePyramidNetwork(channels=128)
      (probability_head): <tensorflow.python.keras.engine.sequential.Sequential object at 0x7fbf100ee5e0>
      (threshold_head): <tensorflow.python.keras.engine.sequential.Sequential object at 0x7fbf101a0b20>
      (postprocessor): DBPostProcessor(box_thresh=0.1, max_candidates=1000)
    )
  )
  (reco_predictor): RecognitionPredictor(
    (pre_processor): PreProcessor(
      (resize): Resize(output_size=(32, 128), method='bilinear', preserve_aspect_ratio=True, symmetric_pad=False)
      (normalize): Compose(
        (transforms): [
          LambdaTransformation(),
          Normalize(mean=[0.5, 0.5, 0.5], std=[1.0, 1.0, 1.0]),
        ]
      )
    )
    (model): CRNN(
      (feat_extractor): <doctr.models.backbones.vgg.VGG object at 0x7fbf1035dfd0>
      (decoder): <tensorflow.python.keras.engine.sequential.Sequential object at 0x7fbf1003f730>
      (postprocessor): CTCPostProcessor(vocab_size=118)
    )
  )
  (doc_builder): DocumentBuilder(resolve_lines=False, resolve_blocks=False, paragraph_break=0.035)
)
```
_Note that postprocessor are now printed and that you have the details of each preprocessor_

Resolves #256 

Any feedback is welcome!

### CODE REVIEW

File: docs/source/models.rst [hunk 1]
```diff
@@ -168,17 +167,17 @@ We used a c5.x12large from AWS instances (CPU Xeon Platinum 8275L) to perform ex
...
-| aws textract                |        N/A      |    75.77   |     77.70     |    70.47   |     69.13     |    46.39   |     43.32     |
-+-----------------------------+-----------------+------------+---------------+------------+---------------+------------+---------------+
++------------------------------------+-----------------+----------------------------+----------------------------+----------------------------+
+|                                    |                 |          Receipts          |            Invoices        |            IDs             |
++====================================+=================+============+===============+============+===============+============+===============+
+| **Architecture**                   | **Input shape** | **Recall** | **Precision** | **Recall** | **Precision** | **Recall** | **Precision** |
++------------------------------------+-----------------+------------+---------------+------------+---------------+------------+---------------+
+| db_resnet50 + crnn_vgg16_bn (ours) | (1024, 1024, 3) | **78.90**  | **81.01**     | 65.68      | **69.86**     | **49.48**  | **50.46**     |
```
Comment (charlesmindee) [inline docs/source/models.rst:175]: NICE!

File: docs/source/models.rst [hunk 2]
```diff
@@ -141,19 +140,19 @@ Predictors that localize and identify text elements in images
 +--------------------------------------------------------------+--------------------------------------+--------------------------------------+
 |                                                              |                  FUNSD               |                  CORD                |
 +=============================+=================+==============+============+===============+=========+============+===============+=========+
-|      **Architecture**       | **Input shape** | **# params** | **Recall** | **Precision** | **FPS** | **Recall** | **Precision** | **FPS** |
+| **Architecture**            | **Input shape** | **# params** | **Recall** | **Precision** | **FPS** | **Recall** | **Precision** | **FPS** |
 +-----------------------------+-----------------+--------------+------------+---------------+---------+------------+---------------+---------+
-| db_resnet50 + crnn_vgg16_bn | (1024, 1024, 3) |              |   0.648    |     0.703     |  0.85   |    0.677   |     0.784     |   1.6   |
+| db_resnet50 + crnn_vgg16_bn | (1024, 1024, 3) |              | 70.08      | 74.77         | 0.85    | 82.19      | 79.67         | 1.6     |
```
Comment (charlesmindee) [inline docs/source/models.rst:145]: maybe we can put in **bold** CORD precision for our model :)

**Review by charlesmindee — CHANGES_REQUESTED** [review]
Thanks for the PR ! Only a small comment on the results

File: docs/source/models.rst (hunk 2 above)
Comment (fg-mindee) [reply docs/source/models.rst:145]: Good idea, I'll fix that!

**Review by charlesmindee — APPROVED** [review]
thanks!

**Review by charlesmindee — APPROVED** [review]
thanks
